src/Groups/matrices/MatGrp.jl

```diff
@@ -478,9 +478,17 @@ Base.getindex(x::MatrixGroupElem, i::Int, j::Int) = x.elm[i,j]
 """
     nrows(x::MatrixGroupElem)
 
-Return the number of rows of the given matrix.
+Return the number of rows of the underlying matrix of `x`.
 """
-nrows(x::MatrixGroupElem) = x.parent.deg
+nrows(x::MatrixGroupElem) = nrows(matrix(x))
+
+"""
+    ncols(x::MatrixGroupElem)
+
+Return the number of columns of the underlying matrix of `x`.
+"""
+ncols(x::MatrixGroupElem) = ncols(matrix(x))
```

Comment:
Hmm.
My understanding of `MatrixGroupElem` is as follows:
The element `x` stores either an Oscar matrix in `x.elm` or a GAP matrix in `x.X` or both.
Thus `matrix(x)` may force the computation of an Oscar matrix from `x.X`.
(Note that `x.elm` calls a suitable `getproperty` method.)

Perhaps it would be good to get rid of the `getproperty` method which somehow hides the fact that the call means more than just a field access, and to call `matrix(x)` in all places where one really needs the Oscar matrix.
Then `x.elm` calls should be restricted to those situations where one is sure that the field is already set.